metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -39,6 +38,13 @@ final class Doctor(
   private val hasProblems = new AtomicBoolean(false)
   private var bspServerName: Option[String] = None
   private var bspServerVersion: Option[String] = None
+  private val problemResolver =
+    new ProblemResolver(
+      workspace,
+      () => bspServerVersion.exists(isUnsupportedBloopVersion),
+      () => bspServerVersion,
```

Comment:
I think the logic is a bit mixed in here and in `ProblemResolver.scala` and it makes it a _bit_ hard to follow.

Do you think it's necessary to actually have `isUnsupportedBloopVersion` _and_ `bloopVersion` be parameters in `ProblemResolver`? What do think of instead just move all of this logic into `ProblemResolver` and handle it all there. We'd move the `isUnsupportedBloopVersion` method right below this into there as well. Then there would be no need to have both of these parameter, just the `BloopVersion`, and then we could do the logic of determining if it's unsupported in the `ProblemResolver`. wdyt?

Reply:
Good idea! I moved everything to `ProblemResolver` I also changed to only use `() => Option[BspSession]` which replaces 3 parameters in the doctor and we no longer need the vars.

Reply:
win, win, win 👍🏼